src/App.tsx

```diff
@@ -25,10 +58,22 @@ function App() {
                 <Analytics />
               </Route>
               <Route path="/">
-                {/* "Your past streams" could be replaced with {regular or twitch username}'s past streams */}
                 <p>Your past streams</p>
                 <StreamsTableView userId="rippyae" />
-              </Route>
+              </Route> */}
+              <UnauthenticatedRoute exact path="/signup" component={SignUp} />
+              <UnauthenticatedRoute exact path="/login" component={Login} />
+              <AuthenticatedRoute
```

Comment:
Problem 2 you mentioned in discord is from here. You can add the route like this (considering that userId is a string and email is `string | null | undefined`)

```ts
<AuthenticatedRoute exact path="/">
    <StreamsTableView userId={email} />
</AuthenticatedRoute>
```

I think what you have passes a function and as such no arguments are passed. ([component vs render](https://v5.reactrouter.com/web/api/Route/component)) Maybe @kevmo314 can help us a bit since I don't really understand if thats whats happening here.

Reply:
You might be able to use `element` here instead: https://ui.dev/react-router-pass-props-to-components/

Reply:
Tried both without luck. `element` isnt available in router v5.  

Is it becuase my initial `createContext(null)` has a value of null? I think right now it's having type issues with `string|null|undefined`

Reply:
Yeah you would have to update `StreamsTableView` to accept and handle `string|null|undefined` or make sure you are passing the `string` it requires right now. (maybe `!` or `??` could help depending on the behaviour of `email`... can it be null or undefined when you are authenticated?)

Reply:
i believe i was able to fix the type errors, but getting the issue of passing a component as a prop for authenticating the home page,  as highlighted by @kevmo314 . His solution only works in react router 6, should we upgrade? Open to other ideas in router v5
![image](https://user-images.githubusercontent.com/13722143/145111958-0207e4fe-fbb1-4e01-a52f-233d45c9c8a0.png)

Reply:
I think the last problem you have is related to what components `<AuthProvider>` is wrapping, since `App` is not wrapped by it I believe that affects your interaction with `useAuthState()`, so if it wrapped `<App/>` then you could use it...

In `index.ts`, something like:
```ts
<React.StrictMode>
    <AuthProvider>
      <App />
    </AuthProvider>
  </React.StrictMode>,
```
And then you should be able to remove `<AuthProvider>` from `App.tsx` and get your email.